This plugins fails to link because of --as-needed.
One (ugly) solution would be to link libpthread and libc in.
Need to find a proper solution.
--- a/akonadi/server/CMakeLists.txt
+++ b/akonadi/server/CMakeLists.txt
@@ -18,7 +18,7 @@
 add_subdirectory( control )
 add_subdirectory( src )
 if (MYSQL_EMBEDDED_FOUND AND OPENSSL_FOUND)
-  add_subdirectory( sqlplugin )
+#  add_subdirectory( sqlplugin )
 endif (MYSQL_EMBEDDED_FOUND AND OPENSSL_FOUND)
 add_subdirectory( tests )
 
